### PR TITLE
JavaScriptTypeRecovery: Replace [Require|JavaScriptCallLinker]Pass

### PR DESCRIPTION
**Pull request by DavidBakerEffendi**
* This PR further overrides and improves upon existing capability in `XTypeHintCallLinker` and `XTypeRecovery` to fully replace the work being done under `RequirePass` and `JavaScriptCallLinkerPass`
* Additionally, `require` imports that are relative are now resolved and a test verifying this has been added
* `:` is now the path separator of `JavaScriptTypeRecovery` to remain consistent with the frontend

### CODE REVIEW

**Comment by DavidBakerEffendi** [comment]
The failing test is strange
```javascript
const foo = (x, y) => {
  sink(x);
};
foo(1, 2);
```
`Identifier(foo) = MethodRef(Test0.js::program:anonymous)` but `Test0.js::program:anonymous` does not exist, it seems to have been replaced by `Test0.js::program:foo`. So my type recovery points `foo` to `Test0.js::program:anonymous`. Will need to make a workaround here.

**Comment by DavidBakerEffendi** [comment]
@max-leuthaeuser More on above, seems like `MethodRef.methodFullName = Test0.js::program:anonymous` but `MethodRef.referencedMethod.fullName = Test0.js::program:foo` so not sure if that is an inconsistency worth fixing since I can just use the latter.

**Comment by max-leuthaeuser** [comment]
Pretty sure that's from @fabsx00 https://github.com/joernio/joern/blob/master/joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/passes/ConstClosurePass.scala

**Comment by DavidBakerEffendi** [comment]
@max-leuthaeuser Ah I see, thanks.

**Review by max-leuthaeuser — APPROVED** [review]

File: joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/JsSrc2Cpg.scala [hunk 1]
```diff
@@ -57,11 +56,19 @@ object JsSrc2Cpg {
...
   def postProcessingPasses(cpg: Cpg, config: Option[Config] = None): List[CpgPassBase] =
     List(
-      new RequirePass(cpg),
       new ConstClosurePass(cpg),
-      new JavascriptCallLinker(cpg),
       new JavaScriptTypeRecovery(cpg, enabledDummyTypes = !config.exists(_.disableDummyTypes)),
       new JavaScriptTypeHintCallLinker(cpg)
     )
 
+  def stripQuotes(str: String): String = {
```
[the remark this answers is not in the document]
Comment (DavidBakerEffendi) [reply joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/JsSrc2Cpg.scala:64]: Seems like `.replaceAll("^\"|'|\"|'$", "")` should do the trick

**Pull request merged**